src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -198,6 +198,26 @@ int sd_dhcp6_client_set_duid(sd_dhcp6_client *client, uint16_t duid_type,
         return 0;
 }
 
+int sd_dhcp6_client_get_duid(sd_dhcp6_client *client, uint16_t *duid_type,
+                             const uint8_t **duid, size_t *duid_len) {
+        assert_return(client, -EINVAL);
+        assert_return(duid_type, -EINVAL);
+        assert_return(duid_len, -EINVAL);
+        assert_return(duid, -EINVAL);
+        assert_return(IN_SET(client->state, DHCP6_STATE_STOPPED), -EBUSY);
+
+        *duid_type = 0;
+        *duid_len = 0;
+        *duid = NULL;
+        if (client->duid_len > 0) {
+                *duid_type = be16toh(client->duid.type);
+                *duid_len = client->duid_len - sizeof(client->duid.type);
+                *duid = client->duid.raw.data;
```

Comment:
The precedent to have sd_dhcp6_client.duid_len hold sizeof(duid.type) + sizeof(duid.type_specific) comes from the original dhcp_identifier_set_duid_en code. Please see: https://github.com/systemd/systemd/blob/master/src/libsystemd-network/dhcp-identifier.c#L34 

So in my earlier commit, when we reverted back to passing in ( uint8_t *duid, size_t duid_len ) in sd_dhcp6_client_set_duid instead of struct duid (which was breaking the public headers), I had to follow the same pattern as dhcp_identifier_set_duid_en (and similar to the IPv4 sd_dhcp_client_set_client_id) to allow the raw.data to be copied-in and add sizeof(duid.type) in sd_dhcp6_client_set_duid. 

This means the get API needs to do the complementary subtract, similar to the IPv4 counterpart: sd_dhcp_client_get_client_id function.